packages/flutter_driver/lib/src/driver/web_driver.dart

```diff
@@ -0,0 +1,270 @@
+    List<TimelineStream> streams = const <TimelineStream>[TimelineStream.all],
+    bool retainPriorEvents = false,
+  }) async {
+    if (_browserName != kChrome) {
+      throw UnimplementedError();
+    }
+    if (!retainPriorEvents) {
+      await clearTimeline();
+    }
+    await startTracing(streams: streams);
+    await action();
+
+    return stopTracingAndDownloadTimeline();
+  }
+
+  @override
+  Future<void> clearTimeline({Duration timeout = kUnusuallyLongTimeout}) async {
+    if (_browserName != kChrome) {
+      throw UnimplementedError();
+    }
+
+    // Reset start time
+    _startTime = null;
+  }
+}
+
+/// Encapsulates connection information to an instance of a Flutter Web application.
+class FlutterWebConnection {
+  FlutterWebConnection._(this._driver);
+
+  final sync.WebDriver _driver;
+
+  /// Starts WebDriver with the given [capabilities] and
+  /// establishes the connection to Flutter Web application.
+  static Future<FlutterWebConnection> connect(String url, Map<String, dynamic> settings) async {
+    final sync.WebDriver driver = createDriver(settings);
+    driver.get(url);
+
+    // Configure WebDriver browser by setting its location and dimension.
+    final List<String> dimensions = settings['browser-dimension'].split(',');
+    if (dimensions.length != 2) {
+      throw DriverError('Invalid browser window size.');
+    }
+    final int x = int.parse(dimensions[0]);
+    final int y = int.parse(dimensions[1]);
+    final sync.Window window = driver.window;
+    window.setLocation(const Point<int>(0, 0));
+    window.setSize(Rectangle<int>(0, 0, x, y));
+
+    // Wait until extension is installed.
+    await waitFor<void>(() => driver.execute('return typeof(window.\$flutterDriver)', <String>[]),
+        matcher: 'function',
+        timeout: const Duration(minutes: 1));
+    return FlutterWebConnection._(driver);
+  }
+
+  /// Sends command via WebDriver to Flutter web application
+  Future<dynamic> sendCommand(String script, Duration duration) async {
+    dynamic result;
+    try {
+      _driver.execute(script, <void>[]);
+    } catch (_) {
+      // In case there is an exception, do nothing
+    }
+
+    try {
+      result = await waitFor<dynamic>(() => _driver.execute('return \$flutterDriverResult', <String>[]),
+          matcher: isNotNull,
+          timeout: duration ?? const Duration(seconds: 30));
+    } catch (_) {
+      // Returns null if exception thrown.
+      return null;
+    } finally {
+      // Resets the result.
+      _driver.execute('''
+        \$flutterDriverResult = null
+      ''', <void>[]);
+    }
+    return result;
+  }
+
+  /// Gets performance log from WebDriver.
+  List<sync.LogEntry> get logs => _driver.logs.get(sync.LogType.performance);
+
+  /// Takes screenshot via WebDriver.
+  List<int> screenshot()  => _driver.captureScreenshotAsList();
+
+  /// Closes the WebDriver.
+  Future<void> close() async {
+    _driver.quit();
```

Comment:
Does driver.quit return a future? If so it should be returned or awaited

Reply:
No it doesn't. It returns void.